src/main/java/de/tum/in/www1/artemis/web/rest/ExamResource.java

```diff
@@ -265,81 +234,72 @@ public ExamResource(UserRepository userRepository, CourseRepository courseReposi
         Exam exam = examRepository.findByIdWithRegisteredUsersExerciseGroupsAndExercisesElseThrow(examId);
         ExamChecklistDTO examChecklistDTO = examService.getStatsForChecklist(exam);
-
         return ResponseEntity.ok(examChecklistDTO);
     }
 
     /**
-     * GET /courses/{courseId}/exams/{examId}/scores : Find scores for an exam by id.
+     * GET courses/{courseId}/exams/{examId}/scores : Find scores for an exam by id.
      *
      * @param courseId              the course to which the exam belongs
      * @param examId                the exam to find
      * @return the ResponseEntity with status 200 (OK) and with the found ExamScoreDTO as body
      */
-    @GetMapping("/courses/{courseId}/exams/{examId}/scores")
+    @GetMapping("courses/{courseId}/exams/{examId}/scores")
     @PreAuthorize("hasRole('INSTRUCTOR')")
     public ResponseEntity<ExamScoresDTO> getExamScore(@PathVariable Long courseId, @PathVariable Long examId) {
         long start = System.currentTimeMillis();
         log.debug("REST request to get score for exam : {}", examId);
-        Optional<ResponseEntity<ExamScoresDTO>> courseAndExamAccessFailure = examAccessService.checkCourseAndExamAccessForInstructor(courseId, examId);
-        if (courseAndExamAccessFailure.isPresent()) {
-            return courseAndExamAccessFailure.get();
-        }
+        examAccessService.checkCourseAndExamAccessForRoleElseThrow(Role.INSTRUCTOR, courseId, examId);
         ExamScoresDTO examScoresDTO = examService.calculateExamScores(examId);
         log.info("get scores for exam {} took {}ms", examId, System.currentTimeMillis() - start);
         return ResponseEntity.ok(examScoresDTO);
     }
 
     /**
-     * GET /courses/:courseId/exams/:examId/exam-for-assessment-dashboard
+     * GET courses/:courseId/exams/:examId/exam-for-assessment-dashboard
      *
      * @param courseId the id of the course to retrieve
      * @param examId the id of the exam that contains the exercises
      * @return data about a course including all exercises, plus some data for the tutor as tutor status for assessment
      */
-    @GetMapping("/courses/{courseId}/exams/{examId}/exam-for-assessment-dashboard")
+    @GetMapping("courses/{courseId}/exams/{examId}/exam-for-assessment-dashboard")
     @PreAuthorize("hasRole('TA')")
     public ResponseEntity<Exam> getExamForAssessmentDashboard(@PathVariable long courseId, @PathVariable long examId) {
-        log.debug("REST request /courses/{courseId}/exams/{examId}/exam-for-assessment-dashboard");
-
+        log.debug("REST request courses/{courseId}/exams/{examId}/exam-for-assessment-dashboard");
         Exam exam = examService.findByIdWithExerciseGroupsAndExercisesElseThrow(examId);
         Course course = exam.getCourse();
         if (!course.getId().equals(courseId)) {
-            return conflict();
+            return badRequest("courseId", "400", "courseId in path and body differ!");
```

Comment:
same here, what is reason using `badRequest `instead of `conflict`? (out of curiosity)